disnake/guild_scheduled_event.py

```diff
@@ -0,0 +1,271 @@
+# The MIT License (MIT)
+
+# Copyright (c) 2021-present DisnakeDev
+
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+from __future__ import annotations
+
+from datetime import datetime
+from typing import Any, Dict, List, Optional, TYPE_CHECKING
+from .enums import (
+    GuildScheduledEventEntityType,
+    GuildScheduledEventStatus,
+    StagePrivacyLevel,
+    try_enum,
+)
+from .user import User
+from .mixins import Hashable
+from .utils import cached_slot_property, parse_time, _get_as_snowflake
+
+if TYPE_CHECKING:
+    from .abc import GuildChannel
+    from .guild import Guild
+    from .state import ConnectionState
+
+
+__all__ = ("GuildEventEntityMetadata", "GuildScheduledEvent")
+
+
+class GuildEventEntityMetadata:
+    """
+    Represents guild event entity metadata.
+
+    .. versionadded:: 2.3
+
+    Attributes
+    ----------
+    speaker_ids: List[:class:`int`]
+        The speakers of the stage channel.
+    location: Optional[:class:`str`]
+        Location of the event
+    """
+
+    __slots__ = ("speaker_ids", "location")
+
+    def __init__(self, *, data: Dict[str, Any]):
+        data = data or {}
```

Comment:
According to the docs (and the typehint) this isn't nullable, not sure if this line is necessary

Reply:
It's null most of the time, the docs are a little bit misleading here

Reply:
I see, maybe the typehint should be `Optional[Dict[str, Any]]` then